File: backend/app/schemas/wallet.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class WalletFeatureRead(BaseModel):
    key: str
    name: str
    description: str
    cost: int
    unlock_package_name: str
    owned: bool


class WalletRead(BaseModel):
    balance: int
    features: list[WalletFeatureRead]


class WalletPurchaseRead(BaseModel):
    id: UUID
    package_id: UUID
    package_name: str
    status: str
    amount_cents: int
    credits_granted: int
    created_at: datetime


class WalletLedgerRead(BaseModel):
    id: UUID
    delta: int
    reason: str
    description: str
    reference_type: str
    reference_id: str
    balance_after: int
    created_at: datetime

File: frontend/src/api/wallet.ts
import client from './client';

export interface WalletFeature {
  key: string;
  name: string;
  description: string;
  cost: number;
  unlock_package_name: string;
  owned: boolean;
}

export interface WalletData {
  balance: number;
  features: WalletFeature[];
}

export interface Purchase {
  id: string;
  package_id: string;
  package_name: string;
  status: string;
  amount_cents: number;
  credits_granted: number;
  created_at: string;
}

export interface LedgerEntry {
  id: string;
  delta: number;
  reason: string;
  description: string;
  reference_type: string;
  reference_id: string;
  balance_after: number;
  created_at: string;
}

export async function getWallet(): Promise<WalletData> {
  const res = await client.get<WalletData>('/api/v1/wallet');
  return res.data;
}

export async function getPurchases(): Promise<Purchase[]> {
  const res = await client.get<Purchase[]>('/api/v1/wallet/purchases');
  return res.data;
}

export async function getLedger(): Promise<LedgerEntry[]> {
  const res = await client.get<LedgerEntry[]>('/api/v1/wallet/ledger');
  return res.data;
}
